src/main/java/org/example/domain/User.java

```diff
@@ -0,0 +1,60 @@
+package org.example.domain;
+
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Version;
+
+@Entity
+public class User {
+
+  @Id
+  private UUID id;
+
+  @Column
+  private String name;
+
+  @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
+  private List<Role> roles;
+
+  @Version
+  private Long version;
+
+  public UUID getId() {
+    return id;
+  }
+
+  public void setId(UUID id) {
+    this.id = id;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public List<Role> getRoles() {
+    return roles;
+  }
+
+  public void setRoles(List<Role> roles) {
+    this.roles = roles;
```

Comment:
fwiw the following change results in 2 roles being persisted (the expected result)

```suggestion
    this.roles.clear();
    this.roles.addAll(roles);
```